pyproject.toml

```diff
@@ -81,6 +81,8 @@ format-check = "ruff format --check {args}"
 
 [tool.hatch.envs.test]
 extra-dependencies = [
+  "numpy>=2",                                  # Haystack is compatible both with numpy 1.x and 2.x, but we test with 2.x
```

Comment:
this pin is not strictly required (see the comment below on `spacy`), but I have included it for clarity